Assets/Resources/UI/Screens/Terrarium/BottomBar/BottomBar.cs

```diff
@@ -2,41 +2,63 @@
 using UnityEngine.UIElements;
 
 public class BottomBar : MonoBehaviour {
-  readonly private string playPauseButtonName = "play-pause-button";
-  readonly private string settingsButtonName = "settings-button";
+  readonly private string gameSpeedDialName = "game-speed-dial";
+  readonly private string gameSpeedButton0Name = "game-speed-0-button";
+  readonly private string gameSpeedButton1Name = "game-speed-1-button";
+  readonly private string gameSpeedButton2Name = "game-speed-2-button";
+  readonly private string gameSpeedButton3Name = "game-speed-3-button";
 
-  readonly private string playString = "Play";
-  readonly private string pauseString = "Pause";
+  readonly private string settingsButtonName = "settings-button";
 
   private UIDocument uiDocument;
-  private Button playPauseButtonVE;
+
+  private VisualElement gameSpeedDialVE;
+  private Button gameSpeedButton0VE;
+  private Button gameSpeedButton1VE;
+  private Button gameSpeedButton2VE;
+  private Button gameSpeedButton3VE;
+
   private Button settingsButtonVE;
 
   private void Awake() {
     uiDocument = GetComponent<UIDocument>();
     VisualElement rootElement = uiDocument.rootVisualElement;
 
-    playPauseButtonVE = rootElement.Q<Button>(playPauseButtonName);
+    gameSpeedDialVE = rootElement.Q<VisualElement>(className: gameSpeedDialName);
+    gameSpeedButton0VE = rootElement.Q<Button>(gameSpeedButton0Name);
+    gameSpeedButton1VE = rootElement.Q<Button>(gameSpeedButton1Name);
+    gameSpeedButton2VE = rootElement.Q<Button>(gameSpeedButton2Name);
+    gameSpeedButton3VE = rootElement.Q<Button>(gameSpeedButton3Name);
     settingsButtonVE = rootElement.Q<Button>(settingsButtonName);
-    GameSpeedManager.OnPauseChanged += KeepPlayPauseButtonNameCorrect;
   }
 
   private void Start() {
     RegisterCallbacks();
   }
 
   private void RegisterCallbacks() {
-    playPauseButtonVE.RegisterCallback<ClickEvent>(
-        (ClickEvent) => { GameSpeedManager.Instance.HandlePause(); });
-    settingsButtonVE.RegisterCallback<ClickEvent>(
-        (ClickEvent) => { TerrariumInputManager.Instance.ToggleSettings(); });
+     gameSpeedButton0VE.RegisterCallback<ClickEvent>(
+          (ClickEvent) => { SetGameSpeedFromDial(0); });
+      gameSpeedButton1VE.RegisterCallback<ClickEvent>(
+          (ClickEvent) => { SetGameSpeedFromDial(1); });
+      gameSpeedButton2VE.RegisterCallback<ClickEvent>(
+          (ClickEvent) => { SetGameSpeedFromDial(2); });
+      gameSpeedButton3VE.RegisterCallback<ClickEvent>(
+          (ClickEvent) => { SetGameSpeedFromDial(3); });
```

Comment:
I decided to just go with 0 based indexing instead of matching the speed just in case we decide to change or have game effects on the speed values later we don't have to change these indeces. Let me know what you think.

Reply:
SetGameSpeedFromDial takes a float which is the speed to set, not an index.

Reply:
Yup you are right! Nice catch.